graphql_compiler/tests/schema_transformation_tests/test_merge_schemas.py

```diff
@@ -456,23 +479,755 @@ def test_clashing_directives(self):
+
+            type Person {
+              identifier: String
+              in_example_edge: [Human] @stitch(source_field: "identifier", sink_field: "id")
+            }
+        ''')
+        self.assertEqual(merged_schema_string, print_ast(merged_schema.schema_ast))
+
+    def test_original_unmodified_when_edges_added(self):
+        basic_schema_ast = parse(ISS.basic_schema)
+        same_field_schema_ast = parse(ISS.same_field_schema)
+        merged_schema = merge_schemas(
+            OrderedDict([
+                ('first', basic_schema_ast),
+                ('second', same_field_schema_ast),
+            ]),
+            [
+                CrossSchemaEdgeDescriptor(
+                    edge_name = 'example_edge',
+                    outbound_side = FieldReference(
+                        schema_id = 'first',
+                        type_name = 'Human',
+                        field_name = 'id',
+                    ),
+                    inbound_side = FieldReference(
+                        schema_id = 'second',
+                        type_name = 'Person',
+                        field_name = 'identifier',
+                    ),
+                    out_edge_only=False,
+                ),
+            ]
+        )
+        self.assertEqual(ISS.basic_schema, print_ast(basic_schema_ast))
+        self.assertEqual(ISS.same_field_schema, print_ast(same_field_schema_ast))
+
+    def test_one_directional_cross_schema_edge_descriptor(self):
+        merged_schema = merge_schemas(
+            OrderedDict([
+                ('first', parse(ISS.basic_schema)),
+                ('second', parse(ISS.same_field_schema)),
+            ]),
+            [
+                CrossSchemaEdgeDescriptor(
+                    edge_name = 'example_edge',
+                    outbound_side = FieldReference(
+                        schema_id = 'first',
+                        type_name = 'Human',
+                        field_name = 'id',
+                    ),
+                    inbound_side = FieldReference(
+                        schema_id = 'second',
+                        type_name = 'Person',
+                        field_name = 'identifier',
+                    ),
+                    out_edge_only=True,
+                ),
+            ]
+        )
+        merged_schema_string = dedent('''\
+            schema {
+              query: RootSchemaQuery
+            }
+
+            type RootSchemaQuery {
+              Human: Human
+              Person: Person
+            }
+
+            type Human {
+              id: String
+              out_example_edge: [Person] @stitch(source_field: "id", sink_field: "identifier")
+            }
+
+            type Person {
+              identifier: String
+            }
+        ''')
+        self.assertEqual(merged_schema_string, print_ast(merged_schema.schema_ast))
+
+    def test_multiple_fields_cross_schema_edge_descriptor(self):
+        multiple_fields_schema = dedent('''\
```

Comment:
Can we move this string to `input_schema_strings.py`, or is there some potential issue I've overlooked?